BDArmory/UI/BDAEditorArmorWindow.cs

```diff
@@ -120,29 +124,30 @@ private void FillArmorList()
         private void OnEditorShipModifiedEvent(ShipConstruct data)
         {
             if (data is null) return;
-            delayedRefreshVisuals = true;
+            //delayedRefreshVisuals = true;
             if (!delayedRefreshVisualsInProgress)
                 StartCoroutine(DelayedRefreshVisuals(data));
         }
 
-        private bool delayedRefreshVisuals = false;
+        //private bool delayedRefreshVisuals = false;
         private bool delayedRefreshVisualsInProgress = false;
         IEnumerator DelayedRefreshVisuals(ShipConstruct ship)
         {
             delayedRefreshVisualsInProgress = true;
             var wait = new WaitForFixedUpdate();
-            while (delayedRefreshVisuals) // Wait until ship modified events stop coming.
-            {
-                delayedRefreshVisuals = false;
+            //while (delayedRefreshVisuals) // Wait until ship modified events stop coming. //preventing the remainder of the ienumerator from getting called. reverting to a simpler wai implementation  -SI
```

Comment:
Was this never escaping the `while` loop? That should only happen if there's a continuous stream of `OnEditorShipModifiedEvent`s every fixed update, which wasn't happening before.

Reply:
Apparently. I was testing by adding a single part to a test ship, and removing it; old way would get caught by the delayedRefreshVisuals bool, according to debug.log lines I stuck in there for testing

Reply:
OK, I'll have to look into why it was doing that then. It shouldn't be getting a continuous stream of `OnEditorShipModifiedEvents`.